powerline/segments/i3wm.py

```diff
@@ -115,3 +115,37 @@ def mode(pl, segment_info, names={'default': None}):
 	if mode in names:
 		return names[mode]
 	return mode
+
+
+def scratchpad_groups(w):
+	group = []
+	if w.urgent:
+		group.append('s_urgent')
+	if w.nodes[0].focused:
+		group.append('s_focused')
+	if w.workspace().name != '__i3_scratch':
+		group.append('s_visible')
+	group.append('scratchpad')
+	return group
+
+
+def scratchpad(pl, fresh_icon=u'\u25cf', changed_icon=u'\u25cb'):
```

Comment:
Ah, and never use `\u…` notation for strings, if you search the whole repository for `\uXXXX` (e.g. `ag '\\u[0-9a-fA-F]{4}'` you will see it only in regexes and tests). This is not readable and the whole point of powerline is to present something to user, thus icons that are more readable in `\uXXXX` notation may not be accepted.